aws-codeartifact-repository/src/main/java/software/amazon/codeartifact/repository/Translator.java

```diff
@@ -237,26 +241,29 @@ static DisassociateExternalConnectionRequest translateDisassociateExternalConnec
         .build();
   }
 
-
-  public static Set<String> translateExternalConnectionFromRepoDescription(RepositoryDescription repository) {
-    return streamOfOrEmpty(repository.externalConnections())
-        .map(RepositoryExternalConnectionInfo::externalConnectionName)
-        .collect(Collectors.toSet());
-  }
-
   /**
    * Request to update Repository
-   * @param model resource model
+   * @param desiredModel desired resource model
+   * @param prevModel previous resource model
    * @return awsRequest the aws service request to modify a resource
    */
-  static UpdateRepositoryRequest translateToUpdateRepository(final ResourceModel model) {
-    return UpdateRepositoryRequest.builder()
-        .domain(model.getDomainName())
-        .domainOwner(model.getDomainOwner())
-        .repository(model.getRepositoryName())
-        .description(model.getDescription())
-        .upstreams(translateToUpstreamList(model))
-        .build();
+  static UpdateRepositoryRequest translateToUpdateRepository(
+      final ResourceModel desiredModel,
+      final ResourceModel prevModel
+  ) {
+    UpdateRepositoryRequest.Builder builder = UpdateRepositoryRequest.builder()
+        .domain(desiredModel.getDomainName())
+        .domainOwner(desiredModel.getDomainOwner())
+        .repository(desiredModel.getRepositoryName())
+        .description(desiredModel.getDescription());
+
+    if (!Objects.equals(prevModel.getUpstreams(), desiredModel.getUpstreams())) {
```

Comment:
are these upstreams lists or sets? if they are lists, this won't work because order matters in list .equals. if it is a set, then it doesnt matter.

Reply:
yes, good call out. I'll update so that order does not matter